monitor/main.go

```diff
@@ -54,21 +75,35 @@ func blockProcessor(ctx context.Context, headers <-chan *types.Header, cc *clien
 		case h = <-headers:
 		}
 
-		logger.Info("Got new header")
+		logHeader(logger, h)
 
-		marshalledHeader, err := json.Marshal(h)
-		if err != nil {
-			return err
-		}
-		fmt.Println(string(marshalledHeader))
 		opts := &bind.CallOpts{
 			BlockNumber: h.Number,
 			Context:     ctx,
 		}
 
+		header, err := cc.Eth.HeaderAndTxnHashesByHash(ctx, h.Hash())
```

Comment:
this a bit unfortunate....  the go-ethereum rpc already sends txHashes, but the impl of EthClient doesn't unmarshall them. In fact this method was created by us. 

A future improvement woudl be to modify EthClient to return the hashes on the ethsubscribe.